hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java

```diff
@@ -122,28 +123,46 @@ public Pair<String, Boolean> process(OMUpdateEventBatch events) {
     final Collection<String> taskTables = getTaskTables();
 
     while (eventIterator.hasNext()) {
-      OMDBUpdateEvent<String, OmKeyInfo> omdbUpdateEvent = eventIterator.next();
+      OMDBUpdateEvent<String, ?> omdbUpdateEvent = eventIterator.next();
       // Filter event inside process method to avoid duping
       if (!taskTables.contains(omdbUpdateEvent.getTable())) {
         continue;
       }
       String updatedKey = omdbUpdateEvent.getKey();
-      OmKeyInfo omKeyInfo = omdbUpdateEvent.getValue();
+
+      // Get the updated and old OM Key Info objects
+      Object omKeyInfo = omdbUpdateEvent.getValue();
+      Object oldOmKeyInfo = omdbUpdateEvent.getOldValue();
+
+      OmKeyInfo keyInfo, oldKeyInfo;
+      // Handle the case where the updated OM Key Info is a RepeatedOmKeyInfo object
+      if (omKeyInfo instanceof RepeatedOmKeyInfo) {
+        // Handle RepeatedOmKeyInfo object
+        RepeatedOmKeyInfo repeatedKeyInfo = (RepeatedOmKeyInfo) omKeyInfo;
+        keyInfo = repeatedKeyInfo.getOmKeyInfoList().get(0);
+        oldKeyInfo = repeatedKeyInfo.getOmKeyInfoList().get(0);
```

Comment:
does it guarantee repeatedKeyInfo has only one key? Also why is keyInfo and oldKeyInfo the same? typo?

Reply:
Thanks for the comment @jojochuang 
I was mistaken in believing that all `OmKeyInfo` objects stored in the `RepeatedOmKeyInfo` class are similar in terms of size, but differ only in their paths. In fact, it is the complete opposite. Once a key is deleted, it is moved to the "om metadata deletedTable." Hence having a mapping of `{label: List<OMKeyInfo>}` ensures that if users repeatedly create and delete keys with the exact same URI, all instances of deletion are grouped together under the same key name.
Hence I believe we must iterate through the list and call `handleDeleteKeyEvent()` on each OmKeyInfo object in the list. Do let me know if I am making a mistake in my understanding of RepeatedOmKeyInfo class and the rocksDB DeletedTable.

Reply:
Something is confusing me.  The fileSizeCount task should only be processing events from the file table and the key table.  They both have value type of OmKeyInfo.  The only table that has RepeatedOmKeyInfo values is the deletedTable.  But those should be skipped here:  https://github.com/apache/ozone/blob/c4494155e78f3686f8c2368b01b7e36c2d4b3a03/hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java#L129

What am I missing?

Reply:
@GeorgeJahad Just saw your comment here. I agree with you (to be double checked with UT/experiments). I have a short writeup [below](https://github.com/apache/ozone/pull/4266#issuecomment-1430514611).

Reply:
@smengcl Don't you think this check would prevent the deleted table from being referenced? and it shouldn't be possible for RepeatedOmKeyInfo from ever being thrown up?  https://github.com/apache/ozone/blob/59938f90bf035dde1285294d4103a49fb0840a8a/hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java#L127

Reply:
@ArafatKhan2198 Looks like it is correctly limiting the events to `keyTable`.

Then we need to find out how those `deletedTable` events could have gotten through this check in the first place.

Any chance for a quick repro UT? Generate an event from `deletedTable` and see if the filter logic is working?

Reply:
I wonder if it could have set the wrong table from the source: https://github.com/apache/ozone/blob/59938f90bf035dde1285294d4103a49fb0840a8a/hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OMDBUpdatesHandler.java#L110

Reply:
@smengcl I made modifications to the UTs for the process in the corresponding test class for `filesizecount`, and the filter check successfully prevented deletedTable from going through.

Reply:
@ArafatKhan2198 So the possibility of `OMDBUpdatesHandler` setting the wrong table above still exists. We need to double check its logic.

@jojochuang Do you have an idea how to repro this? Would doing hsync alone trigger the exception on Recon?